lib/views/push-pull-view.js

```diff
@@ -2,37 +2,196 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 
+import {RemotePropType, BranchPropType} from '../prop-types';
+
+
+function getIconClass(icon, isSyncing = false) {
+  return cx(
+    'github-PushPull-icon',
+    'icon',
+    isSyncing ? 'icon-sync' : `icon-${icon}`,
+    {'animate-rotate': isSyncing},
+  );
+}
+
 export default class PushPullView extends React.Component {
   static propTypes = {
-    pushInProgress: PropTypes.bool,
-    fetchInProgress: PropTypes.bool,
+    currentBranch: BranchPropType.isRequired,
+    currentRemote: RemotePropType.isRequired,
+    isSyncing: PropTypes.bool,
     behindCount: PropTypes.number,
     aheadCount: PropTypes.number,
+    push: PropTypes.func.isRequired,
+    pull: PropTypes.func.isRequired,
+    fetch: PropTypes.func.isRequired,
+    originExists: PropTypes.bool,
   }
 
   static defaultProps = {
-    pushInProgress: false,
-    fetchInProgress: false,
+    isSyncing: false,
     behindCount: 0,
     aheadCount: 0,
   }
 
+
+  componentDidMount() {
+    this.setTooltip();
+  }
+
+  componentDidUpdate() {
+    this.tooltip.dispose();
+    this.setTooltip();
+  }
+
+  setTooltip = () => {
+    this.tooltip = atom.tooltips.add(this.tileNode, {
+      title: `<div style="text-align: left; line-height: 1.2em;">${this.tooltipTarget.dataset.tooltip}</div>`,
+      delay: {show: 200, hide: 100},
+      html: true,
+    });
+  }
```

Comment:
🚨 I'd prefer it if we used the [`<Tooltip>` component](https://github.com/atom/github/blob/master/lib/views/tooltip.js) in render methods rather than manipulating `atom.tooltip` directly in lifecycle methods. It helps centralize the imperative Atom API logic in well-defined, centralized components.